sdk/scheduler/src/main/java/com/mesosphere/sdk/state/GoalStateOverride.java

```diff
@@ -0,0 +1,61 @@
+package com.mesosphere.sdk.state;
+
+import java.util.Arrays;
+
+import com.mesosphere.sdk.specification.GoalState;
+
+/**
+ * The definition of a goal state override for a task. These are custom states, other than {@link GoalState}s, which a
+ * task may enter. The main difference between this and {@link GoalState} is that {@link GoalState}s are configured by
+ * the service developer, whereas {@link GoalStateOverride}s are applied by the operator.
+ */
+public enum GoalStateOverride {
+
+    /** The definition of the default no-override state. Refer to the task's {@link GoalState} setting. */
+    NONE(null, null, null),
+    /** The definition of the "STOPPED" override state, where commands are replaced with sleep()s.*/
+    STOPPED("stop", "STOPPED", "STOPPING");
+
+    private final String command;
+    private final String serializedName;
+    private final String pendingName;
+
+    private GoalStateOverride(String command, String serializedName, String pendingName) {
+        // Validation: Overrides may not overlap with existing GoalStates
+        for (GoalState goalState : GoalState.values()) {
```

Comment:
Hm, since this check is static, I'm loath to spend runtime evaluating it, is there another way to achieve this?

Reply:
FWIW this code is effectively run statically at VM initialization via the "constructor" calls for each of the enum values. So any validation problems should be caught quickly via e.g. a unit test.